src/connectivity/mod.rs

```diff
@@ -267,11 +270,12 @@ pub fn node_connected_component(graph: &graph::PyGraph, node: usize) -> PyResult
         ));
     }
 
-    Ok(conn_components::bfs_undirected(
-        &graph.graph,
-        node,
-        &mut graph.graph.visit_map(),
-    ))
+    let return_value: HashSet<usize> =
+        connectivity::bfs_undirected(&graph.graph, node, &mut graph.graph.visit_map())
+            .into_iter()
+            .map(|x| x.index())
+            .collect();
+    Ok(return_value)
```

Comment:
```suggestion
    Ok(connectivity::bfs_undirected(&graph.graph, node, &mut graph.graph.visit_map())
            .into_iter()
            .map(|x| x.index())
            .collect())
```

Reply:
Done in https://github.com/Qiskit/retworkx/pull/595/commits/d5e5def3556d2221478c59cd334accb372573739.